skbio/tree/_nj.py

```diff
@@ -175,10 +175,11 @@ def _compute_q(dm):
     """
     q = np.zeros(dm.shape)
     n = dm.shape[0]
-    for i in range(n):
-        for j in range(i):
-            q[i, j] = q[j, i] = \
-                ((n - 2) * dm[i, j]) - dm[i].sum() - dm[j].sum()
+    dmv = dm.to_data_frame().values
```

Comment:
Perhaps worth resolving in #1513 since that PR is still open (apologies if it's already resolved there; I haven't taken a close look).